coordinator/coordinator.go

```diff
@@ -490,6 +490,33 @@ func (c *Coordinator) handleMsgSyncBlock(ctx context.Context, msg *MsgSyncBlock)
 	if c.pipeline != nil {
 		c.pipeline.SetSyncStatsVars(ctx, &msg.Stats, &msg.Vars)
 	}
+
+	// If there's any batch not forged by us, make sure we don't keep
+	// "phantom forged l2txs" in the pool.  That is, l2txs that we
+	// attempted to forge in BatchNum=N, where the forgeBatch transaction
+	// failed, but another batch with BatchNum=N was forged by another
+	// coordinator successfully.
+	externalBatchNums := []common.BatchNum{}
+	for _, batch := range msg.Batches {
+		if batch.Batch.ForgerAddr != c.cfg.ForgerAddress {
+			externalBatchNums = append(externalBatchNums, batch.Batch.BatchNum)
+		}
+	}
+	if len(externalBatchNums) > 0 {
+		// If we just synced external batches, make sure the pipeline
+		// is stopped
+		if c.pipeline != nil {
+			if err := c.handleStopPipeline(ctx, "synced external batches", 0); err != nil {
```

Comment:
I have a doubt. Suppose we have 3 external batches from another coordinator and call the `c.handleStopPipeline` without the `failedBatchNum` parameter. It's going to reorg only the last batch, not the last 3 batches. Is this can cause a bug?

https://github.com/hermeznetwork/hermez-node/blob/0e6fc9aa78b711aee0d6bd52cd1c5e4da029b96f/coordinator/coordinator.go#L557-L572

Reply:
Notice that after the call to `handleStopPipeline`, for each external batch there's a call to `l2DB.ResetForged`, which sets all the l2txs with a corresponding batchNum to pending, which is similar to what `l2DB.Reorg` does.

After reading your comment, maybe there's a more clear way to implement this without introducing a new function `l2DB.ResetForged`...

Reply:
I've made an alternative implementation of the same fix here https://github.com/hermeznetwork/hermez-node/pull/720 please, take a look

Reply:
Can we call the reorg only one time passing only the lower block, do we need reorg? Is this can be safer?